common/headtracker/types/head.go

```diff
@@ -0,0 +1,18 @@
+package types
+
+import (
+	txmgrtypes "github.com/smartcontractkit/chainlink/v2/common/txmgr/types"
+	"github.com/smartcontractkit/chainlink/v2/common/types"
+)
+
+type Head[BLOCK_HASH types.Hashable, CHAIN_ID txmgrtypes.ID] interface {
+	types.Head[BLOCK_HASH]
+	// Equals returns true if the two heads are equal
+	Equals(Head[BLOCK_HASH, CHAIN_ID]) bool
```

Comment:
My thought for having the `Equals` method for HeadTracker.Head is because only HeadTracker uses equality comparison. Not sure if this should be under common.Head or HeadTracker.Head.